addons/sale_coupon/models/coupon_program.py

```diff
@@ -12,7 +12,8 @@ class CouponProgram(models.Model):
     # The api.depends is handled in `def modified` of `sale_coupon/models/sale_order.py`
     def _compute_order_count(self):
         for program in self:
-            program.order_count = self.env['sale.order.line'].sudo().search_count([('product_id', '=', program.discount_line_product_id.id)])
+            order_lines = self.env['sale.order.line'].sudo().search([('product_id', '=', program.discount_line_product_id.id)])
+            program.order_count = len(order_lines.mapped('order_id'))
```

Comment:
I think you can use search_count here.
But you may still hear someone yell "don't search in a loop". If that's unpleasant, consider
```python
program_orders = self.env["sale.order.line"].read_group(
    domain=[
        ("order_id.state", "!=", "cancel"), # or not, whatever you end up choosing
        ("product_id", "in", self.mapped("discount_line_product_id").ids),
    ],
    fields=["order_id"],
    groupby=["product_id", "order_id"],
    lazy=False,
)
mapped_data = defaultdict(set)
for value in program_orders:
    mapped_data[value["product_id"][0]] |= {value["order_id"][0]}
for program in self:
    program.order_count = len(mapped_data[program.discount_line_product_id.id])
```
@poma-odoo :eyes:

Reply:
> But you may still hear someone yell "don't search in a loop". If that's unpleasant, consider
This is not just about feeling pleasant or unpleasant, it can also mean an upgrade passes or fails by timeout...

Reply:
@poma-odoo to each their priorities :smile:

Reply:
@odoostan I made the change that you suggested by storing the mapped_date and just omitted the cancel since the user can just filter out the cancelled orders